File: Assets/Second/StatisticsfromaLargeSample.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;

public class StatisticsfromaLargeSample : MonoBehaviour
{
    // Start is called before the first frame update
    void Start()
    {
        int sum = 177847815;
        int total = 1000000;
        Debug.Log(Math.Round(sum / (double)total,5, MidpointRounding.ToEven));
        double[] temp = new double[2];
        temp[0] = sum / (double)total;
        Debug.Log(temp[0]);
    }

    // Update is called once per frame
    void Update()
    {
        
    }

    /**
    https://leetcode.com/problems/statistics-from-a-large-sample/

    计算最大值，最小值，平均数，中位数，众数
    本题c#解法leetcode不能接受，请使用python
     */
    public double[] SampleStats(int[] count) {
        double[] result = new double[5];
        int totalCount = 0;
        int sum = 0;
        int singleMax = 0;
        // 首先找出最大最小
        for(int i = 0; i < count.Length; i++){
            if(result[0] == 0 && count[i] != 0){
                result[0] = (double)i;
            }
            
            if(count[i] != 0){
                result[1] = (double)i;
                totalCount += count[i];
                sum = sum + i * count[i];
                
                if(count[i] > singleMax){
                    result[4] = (double)i;
                    singleMax = count[i];
                }
            }
            
        }
        
        if(count[0] != 0){
            result[0] = (double)0;
        }
        
        result[2] = (double)Math.Round(sum / (decimal)totalCount,6);
        bool isOdd = totalCount % 2 != 0;
        int middleIndex = totalCount / 2;
        int currentCount = 0;
        Console.WriteLine(isOdd);
        for(int i = 0; i < count.Length; i++){
            if(count[i] != 0){
                currentCount += count[i];
            }
            
            if(isOdd && currentCount >= middleIndex){
                result[3] = (double)i;
                return result;
            }else if(!isOdd && currentCount > middleIndex && result[3] == 0){
                result[3] = (double)i;
                return result;
            }else if(!isOdd && currentCount == middleIndex && result[3] == 0){
                result[3] += (double)i;
            }else if(!isOdd && result[3] != 0 && count[i] != 0){
                result[3] += (double)i;
                result[3] = result[3] / 2;
                return result;
            }
        }
        return result;
    }
}

File: PythonSolution/StatisticsfromaLargeSample.py
import bisect

class Solution:
    def sampleStats(self, count):
        result = [0] * 5
        totalCount = 0
        s = 0
        singleMax = 0
        for i in range(len(count)):
            if(result[0] == 0 and count[i] != 0 and i != 0):
                result[0] = i / 1
            
            if(count[i] != 0):
                result[1] = i / 1
                totalCount += count[i]
                s = s + i * count[i]
                
                if(count[i] > singleMax):
                    result[4] = i / 1
                    singleMax = count[i]
        
        if(count[0] != 0):
            result[0] = 0 / 1
        
        result[2] = s / totalCount
        isOdd = totalCount % 2 != 0
        middleIndex = totalCount // 2
        currentCount = 0
        
        for i in range(len(count)):
            currentCount += count[i]
            
            if(isOdd and currentCount >= middleIndex):
                result[3] = i / 1
                return result
            elif(not isOdd and currentCount > middleIndex and result[3] == 0):
                result[3] = i / 1
                return result
            elif(not isOdd and currentCount == middleIndex and result[3] == 0):
                result[3] += i
            elif(not isOdd and result[3] != 0 and count[i] != 0):
                result[3] += i
                result[3] = result[3] / 2
                return result
            
        return result

    def sampleStats_2(self, count):
        n = sum(count)
        mi = next(i for i in range(256) if count[i]) * 1.0
        ma = next(i for i in range(255, -1, -1) if count[i]) * 1.0
        mean = sum(i * v for i, v in enumerate(count)) * 1.0 / n
        mode = count.index(max(count)) * 1.0
        for i in range(255):
            count[i + 1] += count[i]
        median1 = bisect.bisect(count, (n - 1) / 2)
        median2 = bisect.bisect(count, n / 2)
        median = (median1 + median2) / 2.0
        return [mi, ma, mean, median, mode]